back/spec/models/static_page_spec.rb

```diff
@@ -58,6 +74,19 @@
     end
   end
 
+  describe 'before validate' do
+    subject(:static_page) { create(:static_page, projects_filter_type: 'topics', code: 'faq', topics: [topic]) }
+
+    let(:topic) { create(:topic) }
+
+    it 'destroys unused associations' do
+      expect(static_page.topics).to be_present
+      static_page.update!(projects_filter_type: 'areas', areas: [build(:area)])
+      expect(static_page.topics).to be_empty
+      expect(topic).to be_persisted
+    end
+  end
+
```

Comment:
We could add this test. Useful? Thoughts?
```
describe 'projects_filter_type enum' do
    it 'is not valid with a bad projects filter type' do
      expect { build(:static_page, projects_filter_type: 'not_a_valid_enum_value') }
        .to raise_error(ArgumentError)
        .with_message(/is not a valid projects_filter_type/)
    end
  end
  ```

Reply:
Good idea, I'll do it.